src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  length = 'length',
+}
+
+type Props = {
+  sortField: SortType | '';
+  order: boolean;
+};
+
+function getPreparedGoods(goods: string[], { sortField, order }: Props) {
+  const prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [order, setOrder] = useState(false);
+  const listOfGoods = getPreparedGoods(goodsFromServer, { sortField, order });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => setSortField(SortType.alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.length)}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setOrder(!order)}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !order,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || order) && (
+          <button
+            onClick={() => {
+              setOrder(false);
+              setSortField('');
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```